src/instance-id/instance-id.ts

```diff
@@ -0,0 +1,96 @@
+/*!
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {FirebaseApp} from '../firebase-app';
+import {FirebaseInstanceIdError, InstanceIdClientErrorCode} from '../utils/error';
+import {FirebaseServiceInterface, FirebaseServiceInternalsInterface} from '../firebase-service';
+import {FirebaseInstanceIdRequestHandler} from './instance-id-request';
+
+import * as utils from '../utils/index';
+import * as validator from '../utils/validator';
+
+/**
+ * Internals of an InstanceId service instance.
+ */
+class InstanceIdInternals implements FirebaseServiceInternalsInterface {
+    /**
+     * Deletes the service and its associated resources.
+     *
+     * @return {Promise<()>} An empty Promise that will be fulfilled when the service is deleted.
+     */
+    public delete(): Promise<void> {
+        // There are no resources to clean up
+        return Promise.resolve(undefined);
+    }
+}
+
+export class InstanceId implements FirebaseServiceInterface {
+    public INTERNAL: InstanceIdInternals = new InstanceIdInternals();
+
+    private app_: FirebaseApp;
```

Comment:
Let's stick with no trailing _ for private variables. This is also the recommended style for ts.

Reply:
That is going to conflict with the `get app()` getter. Is it better to remove that, and expose `app` as a public attribute of this class?